src/scrapers/base-scraper.js

```diff
@@ -62,23 +62,34 @@ class BaseScraper {
       return transactionsResult;
     }
 
-    this.emitProgress(SCRAPE_PROGRESS_TYPES.SCRAPE_SUMMARY);
-    const summaryResult = await this.fetchSummary();
-    if (!summaryResult.success) {
-      return summaryResult;
+    // TODO: Hack, not sure why all scrapers needs to give summary and payments....
```

Comment:
In my scraper I take checking transactions, dollar, Euro accounts and credit card transactions.
why is this separated here to multiple functions and steps?

Reply:
@esakal when you say `future scraper` you mean this library hasn't implemented that yet?

one issue I have with splitting is that I don't want to run the scraper multiple times, I want to login once and retrieve all the information in one go.
Otherwise it might take longer....
How can we solve that?

I think there are two separate concepts here, one is to split in order to execute separately and one to separate for nicer code.